gemrb/plugins/OpenALAudio/OpenALAudio.cpp

```diff
@@ -515,7 +515,7 @@ Holder<SoundHandle> OpenALAudioDriver::Play(StringView ResRef, unsigned int chan
 #ifdef HAVE_OPENAL_EFX_H
 	ieDword efxSetting = core->GetDictionary().Get("Environmental Audio", 0);
 
-	if (efxSetting && hasReverbProperties && (!p.IsZero() || (flags & GEM_SND_RELATIVE))) {
+	if (efxSetting && hasReverbProperties && channel != SFX_CHAN_GUI && channel != SFX_CHAN_NARRATOR) {
```

Comment:
Come to think of it, shouldn't we also check for GEM_SND_RELATIVE, just in the opposite way (not present)? I can't think of any non-spatial sound that should have spatial effects applied. Something like:
```c++
if (efxSetting && hasReverbProperties && ((channel != SFX_CHAN_GUI && channel != SFX_CHAN_NARRATOR) || !(flags & GEM_SND_RELATIVE))) {
```

Reply:
Then again we can also drop the whole channel check? Unless there are some flags set wrong, like the sound set thing in CG, none of these two (GUI, narrator) is spatial? I guess that simply many cases had (0, 0) for the default value.

Reply:
Yeah, we can simplify. I think GEM_SND_RELATIVE casts a bigger net, so that seems better.

Reply:
Not so useful r/n, since a lot of things go through this and `PlayRelative` applies that flag regardless: https://github.com/gemrb/gemrb/blob/d0af7b1c9952e00489cfe0c39056760eb0e8c3dd/gemrb/core/Interface.cpp#L3396-L3402

And this defsound.2da is both GUI and actions.

Reply:
Sure, but isn't the check currently reversed?

Reply:
That's master: 
```cpp
... && (!p.IsZero() || (flags & GEM_SND_RELATIVE))
```
And by then by that `PlayRelative` many sounds turn into GEM_SND_RELATIVE.

Reply:
That's what I'm saying, shouldn't it be `!(flags & GEM_SND_RELATIVE)`? Relative ie. non-spatial.

Reply:
Eh, ok. My understanding was relative ie. relative to the listener position.

Reply:
Took me a second what was going on there. But I guess we have no good alternative to relying on channels: If you click the PCs, they are fixed towards the listener (GEM_SND_RELATIVE) but since they are on the map, we want them to get the filtered effect. Whereas other actors are subject to the camera position.

Reply:
Yeah.

I've also checked how we use GEM_SND_RELATIVE and if I assume its name makes sense, don't we have these bugs? The way AL_SOURCE_RELATIVE works suggests I was wrong and the name is fine, but check these out:

reversed check:
gemrb/plugins/SDLAudio/SDLAudio.cpp:272 <-- plain wrong
gemrb/core/GameScript/GSUtils.cpp:522 <-- perhaps just a wrong comment, this would need more testing
gemrb/plugins/GUIScript/GUIScript.cpp:7459 <-- we're in a shop, why should this be relative? Also no pos

empty pos:
GemRB_VerbalConstant <-- we could it get from the actor trivially ... or not be relative at all
gemrb/plugins/TLKImporter/TLKImporter.cpp:375 <-- without a clear object to attach it to — should probably be non-spatial anyway

Reply:
I propose this: We remove GEM_SND_RELATIVE and introduce GEM_SND_SPATIAL for everything that is depending on the viewport's position _and_ EFX, and GEM_SND_EFX e. g. for PCs and dialog speaker sounds that are not spatial but that are subject to EFX, and fix these findings? Gets rid of this OAL name, too.

That GSUtils.cpp:522 is good to me in how it is thought of, since I'd expect always full volume on my party feedback, even if the PC is off-screen.

Reply:
I agree, was thinking about renaming the constant as well. It'd be easier with NONSPATIAL, since there's fewer users to inspect, I guess, but the name is worse.

Reply:
Any opposition to merging this now and dealing with the rest in a separate issue/PR?

Reply:
Do it. A little broken then but I'm going through it already.

Reply:
It fixes two other things, so that's why I'm keen. :)